### PR TITLE
update fluo apis with table changes

### PR DESCRIPTION
**Pull request by stufisher**
Think these sps are essentially unused (there was work on i24 before Darren left, but unimplemented i think). 

To be merged with:
https://github.com/DiamondLightSource/ispyb-database/pull/53
https://github.com/DiamondLightSource/SynchWeb/pull/162

### CODE REVIEW

**Comment by KarlLevik** [comment]
Your're getting 
> mysql.connector.errors.IntegrityError: 1048 (23000): Column 'data' cannot be null

... because:
> CREATE TABLE `XRFFluorescenceMapping` (
> ...
>  `data` longblob NOT NULL COMMENT 'The actual data',
> ...

... and your test sets data = None.

**Comment by stufisher** [comment]
Yer wishful thinking i could get away without adding some gzipped data. Added...

**Comment by rjgildea** [comment]
Is this PR ready to merge? It is effectively a dependency of #123 as that requires ispyb-database v1.18.1 (see https://github.com/DiamondLightSource/ispyb-api/pull/123/commits/091ceffb2fb854cca7fa5044779e3e8a20fede06).

**Comment by KarlLevik** [comment]
Should there be any methods/sprocs for the new `XFEFluorescenceComposite` table which was in the PR for the two `XRF*` tables used here?

File: tests/test_mxacquisition.py [hunk 1]
```diff
@@ -4,6 +4,20 @@
...
+
+
+def gzip_json(obj):
+    json_str = json.dumps(obj)
+
+    out = io.BytesIO()
+    with gzip.GzipFile(fileobj=out, mode="w") as fo:
+        fo.write(json_str.encode())
+
+    return out.getvalue()
```
Comment (Anthchirp) [inline tests/test_mxacquisition.py:18]: Any particular reason why this is not just
```suggestion
import binascii
import json
import gzip


def gzip_json(obj):
    return gzip.compress(json.dumps(obj).encode())
```

**Review by Anthchirp — APPROVED** [review]

File: tests/test_mxacquisition.py (hunk 1 above)
Comment (stufisher) [reply tests/test_mxacquisition.py:18]: As long as that produces exactly the same binary thats fine (this is read in other places). Dont think id realised you could compress a string directly.

Comment (Anthchirp) [reply tests/test_mxacquisition.py:18]: we'll see if tests still pass :smile: 

(I've checked with simple data structures, so it should be fine)

**Review by rjgildea — APPROVED** [review]
Please could you add a line to HISTORY.rst documenting the changes?

**Comment by Anthchirp** [comment]
This now has a few conflicts that need resolving.

I've split up the tests so we could test against the 1.18.1 schema with the flou-bits ignored.
It looks like you now need a GridInfo record in the test, but there isn't one in the schema example data by default, so you may have to merge the tests back together (but can leave the independent robot test), or add a fixed GridInfo row to the database schema example data.

The latter would be nicer but would take a bit of effort.

I suppose you want this merged for a 6.0 release?

**Comment by stufisher** [comment]
Added a real gridscan to the source data:
https://github.com/DiamondLightSource/ispyb-database/pull/69

**Pull request merged**